packages/flutter_tools/test/general.shard/ios/devices_test.dart

```diff
@@ -1006,111 +1066,125 @@ void main() {
       });
 
       testWithoutContext('installApp() calls ios-deploy', () async {
+        final List<String> args = <String>[
+          iosDeployPath,
+          '--id',
+          deviceId,
+          '--bundle',
+          bundlePath,
+          '--no-wifi',
+        ];
+        final FakeProcessManager fakeProcessManager = FakeProcessManager.list(<FakeCommand>[
+          FakeCommand(command: args),
+        ]);
+        iosDeploy = IOSDeploy(
+          artifacts: mockArtifacts,
+          cache: mockCache,
+          logger: logger,
+          platform: mockPlatform,
+          processManager: fakeProcessManager,
+        );
         when(mockFileSystem.directory(bundlePath)).thenReturn(directory);
         final IOSDevice device = IOSDevice(
-          '123',
+          deviceId,
           name: 'iPhone 1',
           fileSystem: mockFileSystem,
           sdkVersion: '13.3',
           cpuArchitecture: DarwinArch.arm64,
+          logger: logger,
           platform: mockPlatform,
           artifacts: mockArtifacts,
           iosDeploy: iosDeploy,
         );
-        final List<String> args = <String>[
-          iosDeployPath,
-          '--id',
-          device.id,
-          '--bundle',
-          bundlePath,
-          '--no-wifi',
-        ];
-        when(mockProcessManager.start(any, workingDirectory: anyNamed('workingDirectory'), environment: anyNamed('environment'))).
-            thenAnswer((Invocation invocation) {
-              return Future<Process>.value(createMockProcess());
-            });
 
         await device.installApp(mockApp);
-
-        final List<String> invocationArguments = verify(mockProcessManager.start(
-          captureAny,
-          workingDirectory: anyNamed('workingDirectory'),
-          environment: anyNamed('environment'),
-        )).captured.first as List<String>;
-        expect(invocationArguments, args);
       });
 
       testWithoutContext('uninstallApp() calls ios-deploy', () async {
+        final List<String> args = <String>[
```

Comment:
nit: I would either declare this inline with the FakeProcessManager, or otherwise hoist the entire `FakeCommand` declaration here so it is more coherent.

Reply:
Good point, I thought I re-used the `args` var, but it seems I don't.

Reply:
Done, PTAL.